src/PublicApiGenerator.Tool/Program.cs

```diff
@@ -86,50 +89,104 @@ private static void GeneratePublicApi(string? assembly, string? package, string
+                : null;
 
             try
             {
                 // Because we run in different appdomain we can always unload
-                RunDotnet(workingArea, $"run --configuration Release --framework {framework} -- {fullPath} {outputPath} {outputDirectory}", logVerbose);
-
-                logVerbose.WriteLine($"Public API file: {outputPath}");
-                logVerbose.WriteLine();
+                RunDotnet(workingArea, logVerbose,
+                          apiFilePath != null ? null : Console.Out,
+                          "run",
+                          "--configuration", "Release",
+                          "--framework", framework,
+                          "--",
+                          assemblyPath, apiFilePath ?? "-");
             }
             catch (FileNotFoundException)
             {
-                logError.WriteLine($"Unable to find {fullPath}. Consider specifying --assembly");
+                logError.WriteLine($"Unable to find {assemblyPath}. Consider specifying --assembly");
                 throw;
             }
-        }
 
-        private static void RunDotnet(string workingArea, string arguments, TextWriter logVerbose)
-        {
-            logVerbose.WriteLine($"Dotnet arguments: {arguments}");
+            if (outputDirectory == null || apiFilePath == null)
+                return;
+
+            logVerbose.WriteLine($"Public API file: {apiFilePath}");
             logVerbose.WriteLine();
 
+            var destinationFilePath = Path.Combine(outputDirectory, Path.GetFileName(apiFilePath));
+
+            if (File.Exists(destinationFilePath))
+                File.Delete(destinationFilePath);
+            File.Move(apiFilePath, destinationFilePath);
+
+            Console.WriteLine(Path.GetFullPath(destinationFilePath));
+        }
+
+        private static void RunDotnet(string workingArea, TextWriter logVerbose, TextWriter? stdout, params string[] arguments)
+        {
             var psi = new ProcessStartInfo
             {
                 FileName = "dotnet",
-                Arguments = arguments,
                 WorkingDirectory = workingArea,
                 RedirectStandardOutput = true,
                 RedirectStandardError = true
             };
-            var process = Process.Start(psi);
-            process.WaitForExit(10000);
 
-            var output = process.StandardOutput.ReadToEnd();
-
-            logVerbose.WriteLine($"Dotnet output: {output}");
+            logVerbose.WriteLine($"Invoking dotnet with arguments:");
+            foreach (var (i, arg) in arguments.Select((arg, i) => (i + 1, arg)))
+            {
+                logVerbose.WriteLine($"{i,4}. {arg}");
+                psi.ArgumentList.Add(arg);
+            }
             logVerbose.WriteLine();
 
+            using var process = Process.Start(psi);
+
+            static DataReceivedEventHandler
```

Comment:
The local function mixes with the body of the method and looks wild.

Reply:
Hope it looks better after fdf6d1a471429be2b0d9934338c7cdb998929872.